clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -499,8 +503,9 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
         // Invariant: non-explicit control flow (i.e., kernel-mediated) is indicated
         // by markers.
-        const std::string non_explicit_flow_violation_msg =
-            check_for_pc_discontinuity(shard, memref, cur_instr_decoded, expect_encoding);
+        const std::string non_explicit_flow_violation_msg = check_for_pc_discontinuity(
+            shard, memref, shard->prev_instr_, memref.instr.addr, cur_instr_decoded,
```

Comment:
shard->prev_instr_ is not always the actual previous instr. If a signal just ended, we need to use the last instr before the signal. Maybe use last_instr_in_cur_context_?

Reply:
Any signal event causes the callee to relax all its checks, under the assumption that pre-to-signal-to-post checks are done elsewhere, so it won't matter here.  Added a comment.

Looking more closely at that relaxation: it looks way too loose!  It checks last_xfer_marker which could have happened many instruction records ago!  I added a comment there too.  This PR is large enough and not focused on these corner cases: would prefer to leave to #5912 so just leaving comments here mentioning #5912.

Reply:
Went ahead and replaced prev_instr with last_instr_in_cur_context_ to be more correct even though it shouldn't make a visible difference currently.